server/iframe.go

```diff
@@ -222,14 +261,96 @@ func (a *API) authenticate(w http.ResponseWriter, r *http.Request) {
 	http.SetCookie(w, csrfCookie)
 
 	// Redirect to the home page
-	http.Redirect(w, r, "/", http.StatusSeeOther)
+	http.Redirect(w, r, redirectPath, http.StatusSeeOther)
+}
+
+// MessageHasBeenPosted is called when a message is posted in Mattermost. We rely on it to send a user activity notification
+// to Microsoft Teams when a user is mentioned in a message.
+// This is called in a controller Goroutine in the server side so there's no need to worry about concurrency here.
+func (p *Plugin) MessageHasBeenPosted(c *plugin.Context, post *model.Post) {
+	context := map[string]string{
+		"subEntityId": fmt.Sprintf("post_%s", post.Id),
+	}
+
+	jsonContext, err := json.Marshal(context)
+	if err != nil {
+		p.API.LogError("Failed to marshal context", "error", err.Error())
+		return
+	}
+
+	urlParams := url.Values{}
+	urlParams.Set("context", string(jsonContext))
+
+	for _, mention := range extractMentionsFromPost(post) {
+		u, err := p.apiClient.User.GetByUsername(mention)
+		if err != nil {
+			p.API.LogError("Failed to get user", "error", err.Error())
```

Comment:
I don't think this is an error.  Can this be a debug log?  It's valid for a post to have an `@` then some text that is not a username.  e.g.  giving someone's Instagram handle.